Automatically display recent and upcoming events in order on the homepage

kirsten opened the pull request:
Addressing #48 

This changes the homepage so that the 20 most recent/upcoming events are automatically displayed in reverse chronological order. Events that have already passed are still shown, but without any title or date.

Non-events/posts can now be "pinned" at certain positions in this grid using the same `homepage_position` attribute. Events will be filled in around these "pinned" pages/images.

Also removes the "What's Happening" section from the bottom of the homepage, since it's now redundant.

kirsten committed:
Remove 'homepage_position' from all posts

kirsten committed:
Homepage shows most recent 15 events in reverse chrono order

Remove pages and images for now.

kirsten committed:
Remove upcoming events section from homepage

- Remove styles for homepage events list
- Move includes back into events page now that they are no longer used
by multiple pages
- Remove unused js

kirsten committed:
Hide text associated with past events on homepage

Rework event js to be more flexible and just grab the children of the
parent node.

kirsten committed:
Update jekyll to 3.6.2

This includes support for Liquid 4.0, which has some helpful tags.

kirsten committed:
Update description frontmatter from newtown creek img

kirsten committed:
"Pin" images and pages at certain positions in homepage grid

This is nasty, but liquid doesn't have any way I can find to just insert
something into an array at a certain index...

The homepage grabs the collection of all site documents that include the
homepage_position frontmatter attribute, and also assigns a
`pinned_counter` variable.

During each iteration of the for loop for displaying event posts, there
is a condition that uses the counter variable to see if the pinned doc
at its index has a homepage_position attribute equal to the current
index of the forloop. If it does, an additional element with the
"pinned" content is prepended.

It's also necessary to check the collection type in order to render the
correct markup for the pinned content.

kirsten committed:
Remove homepage position from admin for posts

kirsten committed:
Add .ruby-version to project

kirsten commented:
@dgoats This is the homepage change we talked about on Sat. You should be able to see a preview of it at https://deploy-preview-50--the-sunview.netlify.com/ Let me know what you think!

dgoats commented:
Ah. Good God, yes. I like it. Only one change - can we have the events at
the top sort by ascending order  (so Jan 4 before Jan 6)?
Brilliant!
Thanks Kirsten!!
Dylan

On Jan 2, 2018 10:58 PM, "Kirsten Nordine" <notifications@github.com> wrote:

> @dgoats <https://github.com/dgoats> This is the homepage change we talked
> about on Sat. You should be able to see a preview of it at
> https://deploy-preview-50--the-sunview.netlify.com/ Let me know what you
> think!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/luncheonette/the-sunview/pull/50#issuecomment-354933388>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHGxHfdzbmG2qq32UdXnNyQn1_OqQdH1ks5tGvrKgaJpZM4RRUrC>
> .
>

kirsten committed:
Initial pass at using JSON to render homepage content grid

Creating JSON via Jekyll is ugly af, but being able to work with a
JavaScript array makes it so much easier to get fine grained control
over how the grid is shown.

kirsten committed:
Render homepage grid items using React

kirsten committed:
Add webpack production config

kirsten committed:
Remove baseurl from config

kirsten commented:
Okay, should be good to go now! There aren't any more upcoming post at the moment but they should be sorted as follows:

- Upcoming events are sorted first in ascending order, with all details shown
- Past events are sorted in descending order afterwards, with only images
- "Pinned" content (CSA page, random images) is shown at whatever position it is pinned to
